development/adr/0009-versioning-strategy.md

```diff
@@ -0,0 +1,32 @@
+# **ADR-009: Versioning strategy**
+
+## Status
+
+Proposed
+
+## Context
+
+As a software product, we should establish a sense of versioning.
+With versioning we are able to pin our products as a versioned resource at customer facing deployments
+or for feature planing. This ADR is focused on the Git implementation of our versioning strategy.
+
+## Decision
+
+The base of every versioning strategy is the version number. We are using [Semantic Versioning 2.0.0](https://semver.org/) with version prefix of `v`.
+Our Git strategy is based on the branching and versioning strategy of [Dapr](https://github.com/dapr/dapr).
+
+- New changes are made against the `main` branch.
+- For every `MINOR` release, we are opening a branch called `release-v[MAJOR].[MINOR]` (release branch) based of the previous release.
+- New changes for the release branch are either merged or cherry-picked from `main`.
+- The version of the software on a release branch, until tagged, is called pre-release.
```

Comment:
`pre-release` already has a meaning in the semver spec, we should choose something else

Reply:
[The meaning of a `pre-release` is not clear IMO. Everything that is `[0-9A-Za-z-].` after the patch version seperation via `-` is a `pre-release`](https://semver.org/#spec-item-9).

But yes, we should define `pre-releases` more precisely. Can we even find out the upcoming version on the release branch just from new commits? If a commit follows the tag `v1.0.0` on the branch `release-v1.0`, is the newly created commit automatically the pre-release `v1.0.1-rc-[commit-hash]`? (rc = release candidate)

Reply:
I've added `Continues tagging of the pre-release commits is optional and is at the discretion of the particular implementation of the repository.`